app/models/spree/admin/main_menu/default_configuration_builder.rb

```diff
@@ -0,0 +1,221 @@
+                                                  .with_match_path(/state_in/)
+                                                  .build)
+                            .build
+          c.add(s)
+        end
+
+        def add_returns_section(c)
+          s = SectionBuilder.new('returns', 'reply-fill.svg')
+                            .with_manage_ability_check(Spree::ReturnAuthorization, Spree::CustomerReturn)
+                            .with_item(ItemBuilder.new('return_authorizations', admin_return_authorizations_path)
+                                                  .with_manage_ability_check(Spree::ReturnAuthorization)
+                                                  .with_match_path('/return_authorizations')
+                                                  .build)
+                            .with_item(ItemBuilder.new('customer_returns', admin_customer_returns_path)
+                                                  .with_manage_ability_check(Spree::CustomerReturn)
+                                                  .with_match_path('/customer_returns')
+                                                  .build)
+                            .build
+          c.add(s)
+        end
+
+        def add_products_section(c)
+          s = SectionBuilder.new('products', 'tags-fill.svg')
+                            .with_admin_ability_check(Spree::Product)
+                            .with_item(ItemBuilder.new('products', admin_products_path).with_match_path('/products').build)
+                            .with_item(ItemBuilder.new('option_types', admin_option_types_path)
+                                                  .with_admin_ability_check(Spree::OptionType)
+                                                  .with_match_path('/option_types')
+                                                  .build)
+                            .with_item(ItemBuilder.new('properties', admin_properties_path)
+                                                  .with_admin_ability_check(Spree::Property)
+                                                  .build)
+                            .with_item(ItemBuilder.new('prototypes', admin_prototypes_path)
+                                                  .with_admin_ability_check(Spree::Prototype)
+                                                  .build)
+                            .with_item(ItemBuilder.new('taxonomies', admin_taxonomies_path)
+                                                  .with_admin_ability_check(Spree::Taxonomy)
+                                                  .with_match_path('/taxonomies')
+                                                  .build)
+                            .with_item(ItemBuilder.new('taxons', admin_taxons_path)
+                                                  .with_admin_ability_check(Spree::Taxon)
+                                                  .with_match_path('/taxons')
+                                                  .build)
```

Comment:
Maybe this could be improved? One obvious thing is that the `with_item` API is a bit awkward to build long sections. The simplest thing I can think of is extracting the items into a separate collection:
```rb
products_section_items = [
   ...
]
section = SectionBuilder.new('products', 'tags-fill.svg').with_admin_ability_check(Spree::Product)

products_section_items.each do |item|
  section = section.with_item(item)
end
```

This can be improved further with a `with_items` method.

That approach demonstrates that we could maybe separate the section "definition" from the section itself. The approach in this PR is to create explicit APIs for section manipulation (adding section on `Root` and insertion methods on specific `Section` objects). But what about removal etc? If we separate the section from the builder then we'd also give users the option to simply override the sections themself (changing the list of items directly). It also removes a bit of complexity from the builder. WDYT? Is that something we want to investigate?

Reply:
I see removal is handled with availability checks. So both approaches offer similar functionality.

Reply:
There's a separate `MainMenu::Section` class, that you can create on its own. 
`SectionBuilder` is just a builder class that provides some default values and makes it easier to construct the object.

As for removal:
- availability checks are meant for spree/extension, to allow adding a condition that determines when to display the item. This is not meant for removal, but for adding items that e.g. are only displayed when current user has a permission to access them.
- for removal - yes, I think it's a good idea to add an API for that in the `Section` class (and also provide some options for manipulating the existing items). This way developers could e.g. remove redundant items.